torch/nn/functional.py

```diff
@@ -5178,9 +5185,9 @@ def multi_head_attention_forward(
     #
     # reshape q, k, v for multihead attention and make em batch first
     #
-    q = q.contiguous().view(tgt_len, bsz * num_heads, head_dim).transpose(0, 1)
+    q = q.reshape(tgt_len, bsz * num_heads, head_dim).transpose(0, 1)
```

Comment:
So the above changes assures us that q is contiguous? Could we call view() instead of reshape?

Reply:
In the path where we call `in_projection_packed` yes we could do a view. I think it's also fine in the other paths but to be safe in case we add other paths in the future I changed to reshape.

Is `reshape` more expensive than `view` on the CPU side?

Reply:
It does some checking to see if it is safe to call view otherwise it will clone. I think making it view is more explicit.  I think if the projection changes are to assure contiguity of q,k,v at this point of the computation for perf reasons than we should be explicit. Also I pray no new paths are added

Reply:
Got it, changed to view in [ad36237](https://github.com/pytorch/pytorch/pull/93234/commits/ad362375c457220fe51677562d1080fc87bfcf7f)